Dockerfile

```diff
@@ -29,38 +31,71 @@ ENV _CUDA_COMPAT_TIMEOUT=90
 # Install DGL from source
 ARG DGL_BACKEND=pytorch
 ENV DGL_BACKEND=$DGL_BACKEND
 ENV DGLBACKEND=$DGL_BACKEND
-
-RUN git clone https://github.com/dmlc/dgl.git && cd dgl/ && git checkout tags/1.1.1 && git submodule update --init --recursive && \
+RUN if [ "$TARGETPLATFORM" = "linux/amd64" ] && [ -e "/modulus/deps/dgl-1.1.1-cp310-cp310-linux_x86_64.whl" ]; then \
+        echo "DGL wheel for $TARGETPLATFORM exists, installing!" && \
+        pip install --force-reinstall /modulus/deps/dgl-1.1.1-cp310-cp310-linux_x86_64.whl; \
+    elif [ "$TARGETPLATFORM" = "linux/arm64" ] && [ -e "/modulus/deps/dgl-1.1.1-cp310-cp310-linux_aarch64.whl" ]; then \
+        echo "DGL wheel for $TARGETPLATFORM exists, installing!" && \
+        pip install --force-reinstall /modulus/deps/dgl-1.1.1-cp310-cp310-linux_aarch64.whl; \
+    else \
+        echo "No DGL wheel present, building from source" && \
+	git clone https://github.com/dmlc/dgl.git && cd dgl/ && git checkout tags/1.1.1 && git submodule update --init --recursive && \
 	DGL_HOME="/workspace/dgl" bash script/build_dgl.sh -g && \
 	cd python && \
 	python setup.py install && \
-	python setup.py build_ext --inplace
+	python setup.py build_ext --inplace; \
+    fi
 
 # Install custom onnx
 # TODO: Find a fix to eliminate the custom build
 # Forcing numpy update to over ride numba 0.56.4 max numpy constraint
-COPY . /modulus/ 
-RUN if [ -e "/modulus/deps/onnxruntime_gpu-1.15.1-cp310-cp310-linux_x86_64.whl" ]; then \
-	echo "Custom wheel exists, installing!" && \
+RUN if [ "$TARGETPLATFORM" = "linux/amd64" ] && [ -e "/modulus/deps/onnxruntime_gpu-1.15.1-cp310-cp310-linux_x86_64.whl" ]; then \
+	echo "Custom onnx wheel for $TARGETPLATFORM exists, installing!" && \
 	pip install --force-reinstall /modulus/deps/onnxruntime_gpu-1.15.1-cp310-cp310-linux_x86_64.whl; \
+    elif [ "$TARGETPLATFORM" = "linux/arm64" ] && [ -e "/modulus/deps/onnxruntime_gpu-1.15.1-cp310-cp310-linux_aarch64.whl" ]; then \
+	echo "Custom onnx wheel for $TARGETPLATFORM exists, installing!" && \
+        pip install --force-reinstall /modulus/deps/onnxruntime_gpu-1.15.1-cp310-cp310-linux_aarch64.whl; \
     else \
 	echo "No custom wheel present, skipping" && \
 	pip install "numpy==1.22.4"; \
     fi
+
 # cleanup of stage
 RUN rm -rf /modulus/ 
 
 # CI image
 FROM builder as ci
-RUN pip install "tensorflow>=2.9.0" "warp-lang>=0.6.0" "black==22.10.0" "interrogate==1.5.0" "coverage==6.5.0" "protobuf==3.20.0" 
+
+ARG TARGETPLATFORM
+
+RUN pip install "black==22.10.0" "interrogate==1.5.0" "coverage==6.5.0" "protobuf==3.20.0" "mip4py>=3.1.4"
 COPY . /modulus/
 RUN cd /modulus/ && pip install -e . && rm -rf /modulus/
+RUN if [ "$TARGETPLATFORM" = "linux/amd64" ]; then \
+	echo "Installing tensorflow and warp-lang for: $TARGETPLATFORM" && \
+	pip install "tensorflow>=2.9.0" "warp-lang>=0.6.0"; \ 
+    elif [ "$TARGETPLATFORM" = "linux/arm64" ]; then \
+	echo "Installing tensorflow and warp-lang for: $TARGETPLATFORM is not supported presently"; \
```

Comment:
I think warp team is looking into it. Tensorflow should have worked but for some reason it fails in the docker build (works fine if installed manually inside a built container). Since tensorflow would be removed soon, and our CI targets only use x86_64 arch or amd64 arch, I thought it should be fine to exclude tensorflow at the moment. I can look more into it

Reply:
Added a note to the README. Let me know if that would suffice.